hera_mc/cm_hookup.py

```diff
@@ -272,27 +272,28 @@ def get_hookup(self, hpn_list, rev='ACTIVE', port_query='all', exact_match=False
                    force_new_cache=False, force_db=False, force_db_at_date='now'):
         """
         Return the full hookup to the supplied part/rev/port in the form of a dictionary.
-        Unless force_new_cache is True, it will check a local hookup file and return that if current
-        otherwise it will go through the full database to get hookup.
-        Returns hookup dossier entry, a dictionary with the following entries:
-        This only gets the contemporary hookups (unlike parts and connections,
-            which get all.)  That is, only hookups valid at_date are returned.
-            They are therefore only within one hookup_type
+        The data may come from one of two sources:
+        (1) if the (a) part list is consistent with the keys, (b) date is current, and (c) revision
+            hash is current with a local cache file, data from the cache file will be used.
+        (2) if those conditions are not met, or the flags 'force_new_cache' or 'force_db' are True,
+            it will use the database.
+
+        This only gets the contemporary hookups (unlike parts and connections which get all), unless the
+        force_db flag is True and the force_db_at_date is set.
 
         Parameters
         -----------
         hpn_list:  - list/string of input hera part number(s) (whole or first part thereof)
                    - if string == 'cached' it returns the current dict that would be used
                    - if there are any non-hookup-cached items in list, it defaults to force_db
         rev:  the revision number or descriptor
-        port_query:  a specifiable port name to follow or 'all',  default is 'all'.
+        port_query:  a specifiable port name to follow, or 'all',  default is 'all'.
         exact_match:  boolean for either exact_match or partial
         levels:  boolean to include correlator levels (Currently not working.)
-        force_new_cache:  boolean to force a full database read as opposed to checking file
-                          this will also rewrite the cache-file
-        force_db:  boolean to force this to read/write the file to use the supplied values
-                   Setting this makes get_hookup provide specific hookups (mimicking the
-                   action before the cache file option was instituted)
+        force_new_cache:  boolean to force a full database read as opposed to the cache file.
+                          This will also rewrite the cache file in the process
+        force_db:  boolean to force use of the database, rather than the cache file.  This is primarily
+                   needed to force a search if the date is not the current date (see date below)
```

Comment:
So do you need to set both `force_db` and `force_db_at_date` or does setting `force_db_at_date` also set `force_db`? If the only use for `force_db` is for using it with `force_db_at_date`, why not only have the `force_db_at_date` keyword, which had db in the name already.

Reply:
Excellent point - I don't need both.  I fixed that.